dbt/adapters/sql/connections.py

```diff
@@ -92,11 +92,14 @@ def add_query(
             cursor = connection.handle.cursor()
             cursor.execute(sql, bindings)
 
+            result = self.get_response(cursor)
+
             fire_event(
                 SQLQueryStatus(
-                    status=str(self.get_response(cursor)),
+                    status=str(result),
                     elapsed=time.perf_counter() - pre,
                     node_info=get_node_info(),
+                    query_id=result.query_id,
```

Comment:
@colin-rogers-dbt should I remove this in case the result class as defined in the implementing adapter does not include a `query_id` field?

Reply:
We should add a condition here to only call `result.query_id` if the response has it as an attribute

Reply:
never mind, as long as the adapter returns an AdapterResponse this will exist (it will just be set to `None`. I believe all adapters return that via `get_response`